sim/hunter/talents.go

```diff
@@ -1,870 +1,823 @@
+				return
+			}
+			if spell != hunter.AimedShot && spell != hunter.SteadyShot && spell != hunter.ChimeraShot {
+				return
+			}
+
+			dot := psSpell.Dot(result.Target)
+			outstandingDamage := core.TernaryFloat64(dot.IsActive(), dot.SnapshotBaseDamage*float64(dot.NumberOfTicks-dot.TickCount), 0)
+			newDamage := result.Damage * 0.1 * float64(hunter.Talents.PiercingShots)
+
+			dot.SnapshotBaseDamage = (outstandingDamage + newDamage) / float64(dot.NumberOfTicks)
+			psSpell.Cast(sim, result.Target)
+		},
+	})
+}
+
+func (hunter *Hunter) applyImprovedSteadyShot() {
+	if hunter.Talents.ImprovedSteadyShot == 0 {
+		return
+	}
+	hunter.ImprovedSteadyShotAura = hunter.RegisterAura(core.Aura{
+		Label:     "Improved Steady Shot",
+		ActionID:  core.ActionID{SpellID: 53221, Tag: 1},
+		Duration:  time.Second * 8,
+		MaxStacks: 1,
+		OnGain: func(aura *core.Aura, sim *core.Simulation) {
+			attackspeedMultiplier := 1 + (float64(hunter.Talents.ImprovedSteadyShot) * 0.05)
+			aura.Unit.MultiplyRangedSpeed(sim, attackspeedMultiplier)
+		},
+		OnExpire: func(aura *core.Aura, sim *core.Simulation) {
+			attackspeedMultiplier := 1 + (float64(hunter.Talents.ImprovedSteadyShot) * 0.05)
+			aura.Unit.MultiplyRangedSpeed(sim, 1/attackspeedMultiplier)
+		},
+	})
+	hunter.ImprovedSteadyShotAuraCounter = core.MakePermanent(hunter.RegisterAura(core.Aura{
+		Label:     "Imp SS Counter",
+		ActionID:  core.ActionID{SpellID: 53221, Tag: 2},
+		MaxStacks: 2,
+		OnSpellHitDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+			if spell.ProcMask.Matches(core.ProcMaskRangedAuto) && spell.ActionID.SpellID == 0 && !spell.Flags.Matches(core.SpellFlagAPL) {
+				return
+			}
+			if spell != hunter.SteadyShot {
+				aura.SetStacks(sim, 1)
+			} else {
+				if aura.GetStacks() == 2 {
+					hunter.ImprovedSteadyShotAura.Activate(sim)
+					aura.SetStacks(sim, 1)
+				} else {
+					aura.SetStacks(sim, 2)
+				}
+			}
+		},
+	}))
+}
+func (hunter *Hunter) applyKillingStreak() {
+	if hunter.Talents.KillingStreak == 0 {
+		return
+	}
+	hunter.KillingStreakAura = hunter.RegisterAura(core.Aura{
+		Label:     "Killing Streak",
+		ActionID:  core.ActionID{SpellID: 82748},
+		Duration:  core.NeverExpires,
+		MaxStacks: 1,
+		OnGain: func(aura *core.Aura, sim *core.Simulation) {
+			hunter.KillCommand.DamageMultiplier /= 1 + (float64(hunter.Talents.KillingStreak) * 0.1)
+			hunter.KillCommand.ApplyCostModifiers(hunter.KillCommand.CurCast.Cost - (float64(hunter.Talents.KillingStreak) * 5))
+		},
+		OnExpire: func(aura *core.Aura, sim *core.Simulation) {
+			hunter.KillCommand.DamageMultiplier /= 1 + (float64(hunter.Talents.KillingStreak) * 0.1)
+			hunter.KillCommand.ApplyCostModifiers(hunter.KillCommand.CurCast.Cost + (float64(hunter.Talents.KillingStreak) * 5))
+		},
+		OnSpellHitDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+			if hunter.Talents.KillingStreak > 0 {
```

Comment:
not needed if. You cant get to this if yo udont have points in killing streak as you already return earlier with the same check